Code Improved and Bug Fixed

TT-talhatariq opened the pull request:
I improved the code readability by changing for loops and switch statements with enhanced for loops and switch statements respectively

Also, there is a bug in the Find.java class. If we replace a word of Text with another one, all the characters who're in Uppercase in NotePad, changed to LowerCase.

TT-talhatariq committed:
Code Improved and Bug Fixed

pH-7 commented:
Thanks for this @TT-talhatariq!! Highly appreciated 👏

pH-7 commented:
@TT-talhatariq Why have you added a bunch of images in `bin/`? And they don't seem to be used anywhere...

pH-7 commented on .idea/.gitignore:
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
```
`.gitignore` in the .idea folder...?

pH-7 commented on .idea/.name:
```diff
@@ -0,0 +1 @@
+notepad
```
Why this? I think we don't need this

TT-talhatariq committed:
final code

TT-talhatariq committed:
final code

TT-talhatariq commented:
Actually, it was because of IntelliJ Idea. Once I opened this project in that IDE, and that IDE created some extra folders and files in it.
Anyways, I resolved these issues.

TT-talhatariq reviewed (commented):
Review this now.

pH-7 commented:
> Actually, it was because of IntelliJ Idea. Once I opened this project in that IDE, and that IDE created some extra folders and files in it.
> Anyways, I resolved these issues.

That looks much cleaner now. Thank you! To avoid that, it could be nice to add `.idea` to the root `.gitignore` file then.

pH-7 commented on src/simplejavatexteditor/Find.java:
```diff
@@ -175,7 +175,7 @@ public void replace() {
 
     public void replaceAll() {
-        txt.setText(txt.getText().toLowerCase().replaceAll(textF.getText().toLowerCase(), textR.getText()));
+        txt.setText(txt.getText().replaceAll(textF.getText() , textR.getText()));
```
Was there a reason why you removed the case-insensitive `selection.toLowerCase`?

TT-talhatariq replied:
Because there is a bug in the replaceAll() method of Find.java class. If we replace the word of Text in NotePad with a word, all the characters who're in Uppercase in NotePad, changed to LowerCase.

Like, let's say there is "Talha is Here" in notePad and I replaced the name "Talha" with "Ali" by using the replace all field.
That statement will first change the text of NotePad "Talha is Here" into "talha is ali".

All Uppercase letters are converted into lower case. 

You can Test that replaceAll() function.

pH-7 reviewed (commented):
@TT-talhatariq Why did you remove all `toLowerCase` ?

pH-7 commented:
@TT-talhatariq With what Java version did you test it?

Could you also update the `SimpleJavaTextEditor.jar` with your version and run it? 
You can also generate easily a new jar file with the following command when you are in src/ directory `jar cmvf ../manifest.mf ../SimpleJavaTextEditor.jar simplejavatexteditor/*.class`

pH-7 commented:
@TT-talhatariq Once my concerns have been addressed, I will be happy to approve your work and merge it 😊

TT-talhatariq committed:
Jar File

TT-talhatariq reviewed (commented):
I explained replaceAll() bug above in detail. If you have further query let me Know.

Also, I updated the Jar file. And I'm using java version 17.

Now, there's no issue of cases. First It even replaces the word "talha" with "Talha" in notepad and vice-versa. But this is not possible now and also bug of all lowercase letters resolved.

TT-talhatariq commented:
There is a bug in the Find.java class. If we replace a word of Text with another one, all the characters who're in Uppercase in NotePad, changed to LowerCase.

pH-7 commented:
> 



> I explained replaceAll() bug above in detail. If you have further query let me Know.
> 
> Also, I updated the Jar file. And I'm using java version 17.
> 
> Now, there's no issue of cases. First It even replaces the word "talha" with "Talha" in notepad and vice-versa. But this is not possible now and also bug of all lowercase letters resolved.

Yes, I've seen it now where you mentioned the bug.

Awesome, thanks very much for doing so @TT-talhatariq 😄

pH-7 commented:
> Also, I updated the Jar file. And I'm using java version 17.

Can you replace the current jar file of the project by yours?

pH-7 commented:
Here :) <img width="464" alt="Screenshot 2021-11-21 at 16 17 08" src="https://user-images.githubusercontent.com/1325411/142751277-3030c9c2-ac80-4ab7-990e-d8dcbfc0b327.png">

In the root directory of the project.

TT-talhatariq commented:
> Can you replace the current jar file of the project by yours?

![image_2021-11-21_181419](https://user-images.githubusercontent.com/67586874/142763324-65c69d1f-5d19-4f87-b84e-8a6ef7137f69.png)

TT-talhatariq commented:
> Here :) <img alt="Screenshot 2021-11-21 at 16 17 08" width="464" src="https://user-images.githubusercontent.com/1325411/142751277-3030c9c2-ac80-4ab7-990e-d8dcbfc0b327.png">
> 
> In the root directory of the project.
![image_2021-11-21_181441](https://user-images.githubusercontent.com/67586874/142763336-21de79a4-8481-481d-9f0b-c852f5ebed6a.png)

TT-talhatariq commented:
> Here :) <img alt="Screenshot 2021-11-21 at 16 17 08" width="464" src="https://user-images.githubusercontent.com/1325411/142751277-3030c9c2-ac80-4ab7-990e-d8dcbfc0b327.png">
> 
> In the root directory of the project.

I already Uploaded the version of my own Jar file by replacing it with the Jar file of the root directory.
Here is the picture of the made changes.

TT-talhatariq commented:
> > Also, I updated the Jar file. And I'm using java version 17.
> 
> Can you replace the current jar file of the project by yours?

![pic](https://user-images.githubusercontent.com/67586874/142763423-7251b194-8c18-49dc-941e-0c9447141ae6.png)
But, I have already replaced my own version of jar file with the jar file of the root directory of this project. Here is the picture of changes in project.

TT-talhatariq commented:
> Yes, I've seen it now where you mentioned the bug.
> 
> Awesome, thanks very much for doing so @TT-talhatariq 😄

Thank you for your appreciation 😁

pH-7 commented:
> > > Also, I updated the Jar file. And I'm using java version 17.
> 
> > 
> 
> > Can you replace the current jar file of the project by yours?
> 
> 
> 
> ![pic](https://user-images.githubusercontent.com/67586874/142763423-7251b194-8c18-49dc-941e-0c9447141ae6.png)
> 
> But, I have already replaced my own version of jar file with the jar file of the root directory of this project. Here is the picture of changes in project.

Oh, I see now 🙂 Somehow I missed it 😄 Thank you for your prompt reply @TT-talhatariq! You rock! 👌
